src/zenml/integrations/vertex/google_credentials_mixin.py

```diff
@@ -0,0 +1,51 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+
+from typing import TYPE_CHECKING, Optional, Tuple
+
+from google.auth import default, load_credentials_from_file
+
+if TYPE_CHECKING:
+    from google.auth.credentials import Credentials
+
+
+class GoogleCredentialsMixin:
```

Comment:
Have you considered making this mixin a pydantic `BaseModel` subclass as well so we don't have to specify the `service_account_path` as an attribute on the `VertexStepOperator/VertexOrchestrator` classes? To my knowledge that would work with pydantic, but maybe there are some arguments against it?

Reply:
You're right. I'll do that.